src/clib/pioc_support.c

```diff
@@ -402,46 +484,122 @@ int check_netcdf(file_desc_t *file, int status, const char *fname, int line)
  * @param line the line number of the netCDF call in the code.
  * @return the error code
  */
-int check_netcdf2(iosystem_desc_t *ios, file_desc_t *file, int status,
+int check_netcdf(iosystem_desc_t *ios, file_desc_t *file, int status,
                   const char *fname, int line)
 {
     int eh = default_error_handler; /* Error handler that will be used. */
     char errmsg[PIO_MAX_NAME + 1];  /* Error message. */
+    int ioroot;
+    MPI_Comm comm;
+    int mpierr = MPI_SUCCESS;
 
     /* User must provide this. */
-    pioassert(fname, "code file name must be provided", __FILE__, __LINE__);
-
-    /* No harm, no foul. */
-    if (status == PIO_NOERR)
-        return PIO_NOERR;
+    assert(ios || file);
+    assert(fname);
 
-    LOG((1, "check_netcdf2 status = %d fname = %s line = %d", status, fname, line));
+    LOG((1, "check_netcdf status = %d fname = %s line = %d", status, fname, line));
 
-    /* Pick an error handler. */
-    if (ios)
-        eh = ios->error_handler;
-    if (file)
+    /* Find the error handler. Error handlers associated with file has
+     * priority over ios error handlers.
```

Comment:
This is not correct, I believe. The file->iosystem always equals ios. So the file does not actually have its own error handler.